### DISCUSSION TITLE
Questions about source code of RandomLighting Op

### DISCUSSION THREAD

**Discussion by TonyTnT** [apache/incubator-mxnet#20274] [Q&A]
Questions about A-calculus source code

When I looked at code of RandomLighting, I noticed that var eig is a constant array, and I wanted to ask what does the values of this array means

https://github.com/apache/incubator-mxnet/blob/f5331576431f5b28b8512f2ff4bb17f0268cd79d/src/operator/image/image_random-inl.h#L1062-L1076

https://github.com/apache/incubator-mxnet/blob/f5331576431f5b28b8512f2ff4bb17f0268cd79d/src/operator/image/image_random-inl.h#L1017-L1050
@pluskid

**Comment by zhreshold** [comment]
It's originated from the alexnet's calculation of imagenet PCA noise: https://stackoverflow.com/questions/43328600/does-anyone-have-the-eigenvalue-and-eigenvectors-for-alexnets-pca-noise-from-th

**Reply by TristonC** [reply]
A little bit of history in MXNet feature request found [here](https://github.com/huangzehao/incubator-mxnet-bk/blob/master/src/io/image_aug_default.cc#L481)
